sdk/cosmosdb/cosmos/test/public/integration/query.spec.ts

```diff
@@ -185,46 +185,69 @@ describe("aggregate query over null value", function (this: Suite) {
 
 describe("Test Index metrics", function (this: Suite) {
   this.timeout(process.env.MOCHA_TIMEOUT || 20000);
-  const collectionId = "testCollection3";
 
   beforeEach(async function () {
     await removeAllDatabases();
   });
 
-  it("validate that index metrics are correct for a single partition query", async function () {
-    const database = await getTestDatabase("index metrics test db");
+  it("validate that index metrics are correct", async function () {
+    const collectionId = "testCollection3";
+    const createdContainerSinglePartition = await setupContainer(
+      "index metrics test db",
+      collectionId,
+      4000
+    );
+    const createdContainerMultiPartition = await setupContainer(
+      "index metrics test db multipartioned",
+      collectionId,
+      12000
+    );
+
+    await validateIndexMetrics(createdContainerSinglePartition, collectionId);
+    await validateIndexMetrics(createdContainerMultiPartition, collectionId);
+  });
+
+  async function validateIndexMetrics(container: Container, collectionId: string) {
+    const doc1 = { id: "myId1", pk: "pk1", name: "test1" };
+    const doc2 = { id: "myId2", pk: "pk2", name: "test2" };
+    const doc3 = { id: "myId3", pk: "pk2", name: "test2" };
+    await container.items.create(doc1);
+    await container.items.create(doc2);
+    await container.items.create(doc3);
+    //  stremeable query
+    const query1 = "SELECT * from " + collectionId + " where " + collectionId + ".name = 'test2'";
+    //  aggregate query
+    const query2 = "SELECT * from " + collectionId + " order by " + collectionId + ".name";
+    const queryList = [query1, query2];
+    const queryOptions: FeedOptions = { populateIndexMetrics: true, maxItemCount: 1 };
+    for (const query of queryList) {
+      const queryIterator = container.items.query(query, queryOptions);
+      while (queryIterator.hasMoreResults()) {
+        const { resources: results, indexMetrics } = await queryIterator.fetchNext();
+
+        if (results === undefined) {
+          break;
+        }
+        assert.notEqual(indexMetrics, undefined);
```

Comment:
Can we add checks on structure of `indexMetrics`. i.e. properties we expect to see in the object.

Reply:
Other than this. LGTM.

Reply:
@v1k1, this has been already verified in the older PR: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/cosmosdb/cosmos/test/public/indexMetrics.spec.ts .
This PR is for supporting index metrics feature for non-streameable/aggregate queries.